posthog/temporal/batch_exports/s3_batch_export.py

```diff
@@ -1147,21 +1150,23 @@ async def _upload_part_with_cleanup(
             raise NoUploadInProgressError()
 
         try:
-            await self.logger.ainfo(
+            self.logger.info(
                 "Uploading file number %s part %s with upload id %s",
                 self.current_file_index,
                 part_number,
                 self.upload_id,
             )
             current_key = self._get_current_key()
-            client = await self._get_s3_client()
+            client = self._s3_client
+            assert client is not None, "No S3 client, is multi-part initialized?"
 
             # Retry logic for upload_part
             response: UploadPartOutputTypeDef | None = None
             attempt = 0
 
+            upload_start = time.time()
```

Comment:
the only problem with moving this out of the loop is that it will include retries, which is not necessarily a bad thing, though it could make the upload speed a bit unreliable